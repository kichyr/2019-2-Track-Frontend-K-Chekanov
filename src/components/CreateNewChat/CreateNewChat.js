import React, { useState } from 'react'
import { useDispatch } from 'react-redux'
import * as plusButtonStyles from './plusButtonStyles.module.css'
import * as CNDialogFormStyles from './createNewChatForm.module.css'
import { addNewChatToChatList } from '../../actions/sendMessage'

// create new Chat in local storage and return created chat
function createNewChat(Topic) {
  let data
  if (localStorage.getItem('DialogList') == null) {
    localStorage.setItem('DialogList', '[]')
    data = []
  } else {
    data = JSON.parse(localStorage.getItem('DialogList'))
  }
  const chatId = data.length
  data.push({ chat_id: chatId, topic: Topic, lastmessage: '' })
  localStorage.setItem('DialogList', JSON.stringify(data))
  let mdata
  if (localStorage.getItem('messages') == null) {
    localStorage.setItem('messages', '{}')
    mdata = {}
  } else {
    mdata = JSON.parse(localStorage.getItem('messages'))
  }
  mdata[chatId] = []
  localStorage.setItem('messages', JSON.stringify(mdata))
  return data[data.length - 1]
}

export function PlusButton({ setHiding }) {
  return (
    <div
      role="button"
      id="isActiveForSetHiding"
      tabIndex={0}
      name="plus_butt"
      className={plusButtonStyles.plusbut}
      onClick={setHiding}
    >
      <div className={plusButtonStyles.horizontal_plus} id="isActiveForSetHiding" />
      <div className={plusButtonStyles.vertical_plus} id="isActiveForSetHiding" />
    </div>
  )
}

function CreateNewDialogForm({ isHide, setHiding }) {
  const dispatch = useDispatch()
  let topicForm = null
  // creating new chat when click on button
  const createChatButt = (e) => {
    e.preventDefault()
    dispatch(addNewChatToChatList(createNewChat(topicForm.value)))
    setHiding(e)
  }

  return (
    <div
      className={CNDialogFormStyles.modal}
      id="isActiveForSetHiding"
      style={isHide ? { display: 'block' } : { display: 'none' }}
      role="button"
      tabIndex={0}
      onClick={setHiding}
    >
      <div className={CNDialogFormStyles.modal_content}>
        <span
          className={CNDialogFormStyles.close}
          id="isActiveForSetHiding"
          role="button"
          tabIndex={0}
          onClick={setHiding}
        >
          &times;
        </span>
        <h2>Create new chat</h2>
        <br />
        <input
          className={CNDialogFormStyles.topic}
          ref={(input) => {
            topicForm = input
          }}
          id="topic_form"
          type="text"
          name="topic"
          placeholder="Chat Topic"
        />
        <button onClick={createChatButt}>
          <span name="create_chat_button" id="isActiveForSetHiding">
            Create new chat
          </span>
        </button>
      </div>
    </div>
  )
}

function CreateChatStuff() {
  const [hidden, setHiding] = useState(false)
  const handleClick = (e) => {
    e.preventDefault()
    e.stopPropagation()
    console.log(e.target)
    if (e.target.id === 'isActiveForSetHiding') hidden ? setHiding(false) : setHiding(true)
  }
  return (
    <>
      <PlusButton setHiding={handleClick} />
      <CreateNewDialogForm isHide={hidden} setHiding={handleClick} />
    </>
  )
}

export default CreateChatStuff
