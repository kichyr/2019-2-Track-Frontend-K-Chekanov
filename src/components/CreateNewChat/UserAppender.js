import React, { useState } from 'react'
import Autosuggest from 'react-autosuggest'
import { HOST } from '../whoami'

async function getSuggestions(value, setState) {
  fetch(HOST + 'users/profile/' + value, {
    crossDomain: true,
    mode: 'cors',
    method: 'GET',
    credentials: 'include',
  })
    .then((response) => {
      return response.json()
    })
    .then((data) => {
      const parsedDate = JSON.parse(data)
      setState({ value: value, suggestions: parsedDate.users })
    })
}

const renderSuggestion = (suggestion) => <div>{`${suggestion.first_name} ${suggestion.last_name}`}</div>

// SEARCH USERS
function UserSuggester({ selectedUsers, setSelectedUsers }) {
  const [state, setState] = useState({
    value: '',
    suggestions: [],
  })
  const onChange = (event, { newValue }) => {
    setState({
      ...state,
      value: newValue,
    })
  }
  // Autosuggest will call this function every time you need to clear suggestions.
  const onSuggestionsClearRequested = () => {
    setState({
      value: '',
      suggestions: [],
    })
  }

  // When suggestion is clicked, Autosuggest needs to populate the input
  // based on the clicked suggestion. Teach Autosuggest how to calculate the
  // input value for every given suggestion.
  const getSuggestionValue = (suggestion) => suggestion.name

  // Autosuggest will call this function every time you need to update suggestions.
  // You already implemented this logic above, so just use it.
  const onSuggestionsFetchRequested = ({ value }) => {
    getSuggestions(value, setState)
  }

  const inputProps = {
    placeholder: 'Type a some part of user name',
    value: state.value,
    onChange: onChange.bind(this),
  }

  const onSuggestionSelected = (e, { suggestion }) => {
    console.log(suggestion)
    setSelectedUsers([...selectedUsers, suggestion])
  }

  return (
    <Autosuggest
      suggestions={state.suggestions}
      onSuggestionsFetchRequested={onSuggestionsFetchRequested}
      onSuggestionsClearRequested={onSuggestionsClearRequested}
      getSuggestionValue={getSuggestionValue}
      renderSuggestion={renderSuggestion}
      inputProps={inputProps}
      onSuggestionSelected={onSuggestionSelected}
    />
  )
}

export default UserSuggester
