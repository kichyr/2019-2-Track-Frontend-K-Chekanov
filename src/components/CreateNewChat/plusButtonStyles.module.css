.plusbut {
    position: absolute;
    right: 3vh;
    bottom: 3vh;
    border-radius: 50%;
    width: 8vh;
    height: 8vh;
    background-color: #0084ff;
    transition: all 0.3s;
  }
.plusbut:hover, .plusbut:active{
    background-color: #1231FB;
    width: 10vh;
    height: 10vh;
    right: 2vh;
    bottom: 2vh;
}

.horizontal_plus {
    position: relative;
    background-color: #FFFFFF;
    width: 50%;
    height: 12.5%;
    left: 25%;
    top: 43.75%;
  }
  .vertical_plus {
    position: relative;
    background-color: #FFFFFF;
    width: 12.5%;
    height: 50%;
    left: 43.75%;
    top: 12.5%;
  }