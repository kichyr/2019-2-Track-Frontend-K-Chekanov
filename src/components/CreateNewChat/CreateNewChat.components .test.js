import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import {PlusButton}  from './CreateNewChat'

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

it('renders with PlusButton', () => {
  const button = shallow(
    <PlusButton />
	);
  expect(button).toMatchSnapshot();
  button.find('[role="button"]').simulate('hover');
  expect(button).toMatchSnapshot()
});

it('renders CreateChatForm', () => {
	const createChatForm = shallow(
    <PlusButton />
	);
  expect(createChatForm).toMatchSnapshot();
})