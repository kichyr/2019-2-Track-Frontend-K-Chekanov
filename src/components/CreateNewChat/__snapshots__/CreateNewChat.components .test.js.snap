// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CreateChatForm 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlusButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          id="isActiveForSetHiding"
        />,
        <div
          id="isActiveForSetHiding"
        />,
      ],
      "className": undefined,
      "id": "isActiveForSetHiding",
      "onClick": undefined,
      "role": "button",
      "tabIndex": 0,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": undefined,
          "id": "isActiveForSetHiding",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": undefined,
          "id": "isActiveForSetHiding",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            id="isActiveForSetHiding"
          />,
          <div
            id="isActiveForSetHiding"
          />,
        ],
        "className": undefined,
        "id": "isActiveForSetHiding",
        "onClick": undefined,
        "role": "button",
        "tabIndex": 0,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": undefined,
            "id": "isActiveForSetHiding",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": undefined,
            "id": "isActiveForSetHiding",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`renders with PlusButton 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlusButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          id="isActiveForSetHiding"
        />,
        <div
          id="isActiveForSetHiding"
        />,
      ],
      "className": undefined,
      "id": "isActiveForSetHiding",
      "onClick": undefined,
      "role": "button",
      "tabIndex": 0,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": undefined,
          "id": "isActiveForSetHiding",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": undefined,
          "id": "isActiveForSetHiding",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            id="isActiveForSetHiding"
          />,
          <div
            id="isActiveForSetHiding"
          />,
        ],
        "className": undefined,
        "id": "isActiveForSetHiding",
        "onClick": undefined,
        "role": "button",
        "tabIndex": 0,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": undefined,
            "id": "isActiveForSetHiding",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": undefined,
            "id": "isActiveForSetHiding",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`renders with PlusButton 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlusButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          id="isActiveForSetHiding"
        />,
        <div
          id="isActiveForSetHiding"
        />,
      ],
      "className": undefined,
      "id": "isActiveForSetHiding",
      "onClick": undefined,
      "role": "button",
      "tabIndex": 0,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": undefined,
          "id": "isActiveForSetHiding",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": undefined,
          "id": "isActiveForSetHiding",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            id="isActiveForSetHiding"
          />,
          <div
            id="isActiveForSetHiding"
          />,
        ],
        "className": undefined,
        "id": "isActiveForSetHiding",
        "onClick": undefined,
        "role": "button",
        "tabIndex": 0,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": undefined,
            "id": "isActiveForSetHiding",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": undefined,
            "id": "isActiveForSetHiding",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
